### DISCUSSION TITLE
Can one mimir rule be captured by different notification policies?

### DISCUSSION THREAD

**Discussion by junjie-landing** [grafana/mimir#2071] [Help and support]
For example, we want to define one mimir rule that can notify different oncall group, etc. Can one mimir rule be captured by different notification policies? 

We tried the following setup. If it work, then we can have different notification policy capture alert with different label. For example `team: team-1` and `team: team-2` should be captured by different notification policies.

```
labels:
  oncall_group: infra
  team: {{ $labels.team }}
```

However it doesn't work, it ends up having error liek this:
```
ERRO[0000] unable parse rules file                       error="yaml: unmarshal errors:\n  line 32: cannot unmarshal !!map into string" file=rules/infra/infra-rules.yaml
mimirtool: error: sync operation unsuccessful, unable to parse rules files: file read error, try --help
```

**Comment by pracucci** [comment]
I think what you want is not changing the rule, but changing your Alertmanager configuration:
https://prometheus.io/docs/alerting/latest/configuration/#route

When an alerting rule is firing, a notification is sent to the Alertmanager. Then it's Alertmanager responsability to send the notification through the configured receivers (notification channels). In the Alertmanager routes, by default the routing stop at the first match (because of default `continue: false`) but if you change it to `continue: true` it will continue to evaluate. If it match another route, it will also send the notification through it.

> ERRO[0000] unable parse rules file                       error="yaml: unmarshal errors:\n  line 32: cannot unmarshal !!map into string" file=rules/infra/infra-rules.yaml

There's apparently a syntax error in the file. I would need the whole `rules.yaml` file if you want us to pinpoint the issue there. You can also check the syntax doc here:
https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/

**Comment by dimitarvdimitrov** [comment]
Have you tried putting the template in quotes? I am not sure if this will allow you to use diffferent notification policies, but I think the immediate error you are seeing is caused by this

```yaml
  team: '{{ $labels.team }}'
```
